sourced/ml/__main__.py

```diff
@@ -33,6 +35,21 @@ def add_parser(name, help_message):
             name, help=help_message, formatter_class=ArgumentDefaultsHelpFormatterNoNone)
 
     # ------------------------------------------------------------------------
+    preprocessing_parser = subparsers.add_parser(
+        "preprocess", help="Convert siva to parquet files with selected fields.")
+    preprocessing_parser.set_defaults(handler=preprocess_source_entry)
+    preprocessing_parser.add_argument("-x", "--mode", choices=Moder.Options.__all__,
+                                      default="file", help="What to extract from repositories.")
+    add_repo2_args(preprocessing_parser)
+    add_dzhigurda_arg(preprocessing_parser)
+    preprocessing_parser.add_argument(
+        "-o", "--output", required=True,
+        help="[OUT] Path to the parquet files with bag batches.")
+    default_fields = ["blob_id", "repository_id", "content", "path", "commit_hash", "uast"]
+    preprocessing_parser.add_argument(
+        "-f", "--fields", action="append", default=default_fields,
+        help="Fields to select from DF to save.")
```

Comment:
Does "DF" stand for "document frequencies" or "data frame"? I think Egor made it an RDD of `Row`-s

Reply:
just prune to `Fields to save.`